eth/downloader/queue.go

```diff
@@ -775,12 +778,16 @@ func (q *queue) DeliverBodies(id string, txLists [][]*types.Transaction, txListH
 		if uncleListHashes[index] != header.UncleHash {
 			return errInvalidBody
 		}
+		if header.WithdrawalsHash != nil && (withdrawalListHashes[index] != *header.WithdrawalsHash) {
+			return errInvalidBody
+		}
```

Comment:
If the `header.WithdrawalsHash` is `nil`, and someone delivers a chunk of withdrawals. Then you will not produce an error here. 
Further down, on line `790`, you will set `result.Withdrawals` to the delivered withdrawals. 
Not sure what the impact is, in the end. 
I don't think we should error, though: IMO the proper way to handle it is to just discard the bogus data.

Reply:
Yeah that makes sense. We should check that delivered withdrawals are also matching the header.

Reply:
How would this be used for dos, any more than we already are? If someone sends us a list of transactions, we _will_ to the same amount of work (calculating the trie root of it)